config/raw_config.go

```diff
@@ -46,6 +48,19 @@ func NewRawConfig(raw map[string]interface{}) (*RawConfig, error) {
 	return result, nil
 }
 
+// Copy returns a copy of this RawConfig, uninterpolated.
+func (r *RawConfig) Copy() *RawConfig {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+
+	result, err := NewRawConfig(r.Raw)
+	if err != nil {
+		panic("copy failed: " + err.Error())
+	}
```

Comment:
I panic here because if the *RawConfig was made once, it shouldn't fail ever again. It is a pure function in that way.